Modified to fix typo (mTc2 -> mTa2) by Christian Tacke


From a854fb5c250fe7b7b17e4e43f7dcb03e63ee1364 Mon Sep 17 00:00:00 2001
From: Roberto Preghenella <preghenella@bo.infn.it>
Date: Sat, 27 Jun 2020 19:46:58 +0200
Subject: [PATCH] Patch for floating point exception in Ropewalk.cc (by
 C.Bierlich)

See discussion in JIRA
https://alice.its.cern.ch/jira/browse/ALIROOT-8488
---
 src/Ropewalk.cc | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/src/Ropewalk.cc b/src/Ropewalk.cc
index e4da529..56bef3b 100644
--- a/src/Ropewalk.cc
+++ b/src/Ropewalk.cc
@@ -103,12 +103,18 @@ void RopeDipole::propagateInit(double deltat) {
   // Dipole end momenta.
   Vec4 pcm = d1.getParticlePtr()->p();
   Vec4 pam = d2.getParticlePtr()->p();
-  double mTc = sqrt(pcm.pT2() + pcm.m2Calc());
-  double mTa = sqrt(pam.pT2() + pam.m2Calc());
-  if (mTc == 0 || mTa == 0)
+  
+  double mTc2 = pcm.pT2() + pcm.m2Calc();
+  double mTa2 = pam.pT2() + pam.m2Calc();
+  
+  if (mTc2 <= 0 || mTa2 <= 0) {
     infoPtr->errorMsg("Error in RopeDipole::propagateInit: Tried to"
-      "propagate a RopeDipoleEnd with mT = 0");
-
+      "propagate a RopeDipoleEnd with mT2 <= 0");
+    return;
+  }
+  double mTc = sqrt(mTc2);
+  double mTa = sqrt(mTa2);
+  
   // New vertices in the lab frame.
   Vec4 newv1 = Vec4(d1.getParticlePtr()->xProd() + deltat * pcm.px() / mTc,
                 d1.getParticlePtr()->yProd() + deltat * pcm.py() / mTc, 0, 0);
